lib/common/xml.c

```diff
@@ -1627,7 +1624,8 @@ log_xml_changes(int log_level, const char *file, const char *function, int line,
                 const char *value = crm_element_value(data, aname);
                 flags = prefix_del;
                 do_crm_log_alias(log_level, file, function, line,
-                                 "%s %s @%s=%s", flags, spaces, aname, value);
+                                 "%s %s @%s=%s", flags,
+                                 (const char *) spaces->str, aname, value);
```

Comment:
Would it be straightforward to get rid of `spaces` altogether, and just use the width approach?

Reply:
More or less yes. In this context, `insert_prefix()` is useful mostly as a convenience wrapper for checking whether `log_xml_option_formatted` is set. We don't really need to append to a buffer here.